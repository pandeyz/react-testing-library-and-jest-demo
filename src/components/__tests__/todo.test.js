import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../../components/todo';

afterEach(() => {
    cleanup();
});

test('Todo 1', () => {
    const todo = { id: 1, title: 'Wake up early', completed: true };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId(`todo-${todo.id}`);
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent(todo.title);
    expect(todoElement).toContainHTML('strike');
});

test('Todo 2', () => {
    const todo = { id: 2, title: 'Do excercise', completed: false };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId(`todo-${todo.id}`);
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent(todo.title);
    expect(todoElement).not.toContainHTML('strike');
});

test('Snapshot Test', () => {
    const todo = { id: 1, title: 'Wake up early', completed: true };
    
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
});