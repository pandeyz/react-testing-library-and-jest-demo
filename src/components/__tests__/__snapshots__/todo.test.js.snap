// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      Wake up early
    </h1>
  </strike>
</div>
`;
